btclightclient/txout_verification.go

```diff
@@ -0,0 +1,229 @@
+package btclightclient
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+
+	// "fmt"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+type Hash256Digest [32]byte
+
+type VerifyStatus int
+
+type SPVProof struct {
+	blockHash   *chainhash.Hash
+	txId        *chainhash.Hash
+	txIndex     uint
+	merkleProof []byte
+}
+
+type PMerkleTree struct {
```

Comment:
I think `ParsedMerkeleTree` or simply `MerkeleTree` will be a better name

Reply:
What is this structure? It's not a tree.

Reply:
This is a partial Merkle tree.